### ISSUE TITLE
Search and noscript

### ISSUE THREAD

**Issue by liara** [comfusion/after-dark#98]
If I disable javascript in firefox on, for example, [hackcabin.com](https://hackcabin.com/search/), I see:

![search](https://0x0.st/sjcC.png)

I'm not sure that's supposed to be rendered like that, or is it? I made that whole div tag invisible, but I guess that's not the correct way to do it:

```
  <div id="search-app" style="display:none">
    <section>
      <noscript>
        Aw-shucks! Search requires use of JavaScript to function properly.
      </noscript>
      <form v-on:submit.prevent class="form" action="{{ "search" | absURL }}">
        <fieldset class="form-group">
          <input v-model="query" id="query" name="s" type="search" class="form-control" maxlength="32" autocomplete="off">
          <div class="help-block">Press <kbd>s</kbd> to focus search input anytime.</div>
        </fieldset>
      </form>
    </section>
    <section v-if="results.length">
      <p><i>Showing results for “{ resultsForSearch }”.</i></p>
      <div id="search-results">
        <article v-for="result in results" itemscope itemtype="http://schema.org/CreativeWork">
          <header itemprop="name">
            <h2 itemprop="name"><a :href="result.item.url">{ result.item.title }</a></h2>
          </header>
          <div v-html=result.item.summary itemprop="description"></div>
          <nav class="readmore"><a itemprop="url" :href="result.item.url">Read More&nbsp;&raquo;</a></nav>
        </article>
      </div>
    </section>
  </div>
  <script>document.getElementById('search-app').style.display='';</script>
  <noscript>
    Aw-shucks! Search requires use of JavaScript to function properly.
  </noscript>
```

Or maybe It's just my browser.

Another thing, It's not a Hugo or After Dark issuse, but in case somebody encounters problems with Netlify, looks like one has to disable asset optimization for Search to work: [link](https://discourse.gohugo.io/t/help-chart-js-in-stylesheet/7887/8)

Besides that, Search works great for me. Also great theme. Thank you for your work.

**Comment by jhabdas** [comment]
Thanks for the issue @liara. Your suggestion to programmatically display the search region when JS is enabled looks good to me. If I were to bake similar functionality into the theme I'd accept a pull for something which builds the app on the fly so no unnecessary DOM is created and, thus, will no need to be hidden. If you'd like to work on a pull for this I'm game to review and critique it.

Thanks for the note about the Netlify compressor. I've seen similar behavior with Siteleaf and would bet dollars to doughnuts it's due to HTML compression running over the `script` block and invalidating the current JS. One way around this is to add semicolons after the end of each statement. But the true fix is in the HTML compressor—it should not invalidate valid code such as mine which follows [Standard JS Rules](https://standardjs.com/rules.html#semicolons) for semicolons.

I'm going to close this as a question. But I'm sure it'll be helpful for the next one who comes along wondering. Thank you so much for sharing. 🙏

**Comment by liara** [comment]
Thanks @jhabdas for the insights. 

I had a look at [vue guide](https://vuejs.org/v2/guide/syntax.html#Raw-HTML) and I think I'll go with v-html on the two lines bothering me (the code showing), looks like the simplest thing to do and looks ok. Or is there a security or other reason that plain text should show insted?

```
<p><i>Showing results for <span v-html="resultsForSearch"></span>.</i></p>
<h2 itemprop="name"><a :href="result.item.url"><span v-html="result.item.title"></span></a></h2>
```

I hope your offer to help me with some future pull request still stands:)

**Comment by jhabdas** [comment]
@liara Very nice. I wasn't aware of [`v-cloak`](https://devdocs.io/vue~2/api/index#v-cloak) until you mentioned it and I'm happy to know it exists now. In light of that I'm going to reopen this issue for tracking purposes. I think we can settle for a `v-cloak` option as opposed to dynamically creating the entire app given it's a Vue framework feature which seems was anticipated for this exact use case. Looking forward to any pull you can provide and happy to have more contributors to After Dark.

**Comment by liara** [comment]
@jhabdas Are you ok with this? 

critical-theme.css

```css
[v-cloak] {
  display: none;
}
```

search.html

```html
 <noscript>
    Aw-shucks! Search requires use of JavaScript to function properly.
  </noscript>
  <div id="search-app">
    <div v-cloak>
      <section>
        <form v-on:submit.prevent class="form" action="{{ "search" | absURL }}">
          <fieldset class="form-group">
            <input v-model="query" id="query" name="s" type="search" class="form-control" maxlength="32" autocomplete="off">
            <div class="help-block">Press <kbd>s</kbd> to focus search input anytime.</div>
          </fieldset>
        </form>
      </section>
      <section v-if="results.length">
        <p><i>Showing results for “{ resultsForSearch }”.</i></p>
        <div id="search-results">
          <article v-for="result in results" itemscope itemtype="http://schema.org/CreativeWork">
            <header itemprop="name">
              <h2 itemprop="name"><a :href="result.item.url">{ result.item.title }</a></h2>
            </header>
            <div v-html=result.item.summary itemprop="description"></div>
            <nav class="readmore"><a itemprop="url" :href="result.item.url">Read More&nbsp;&raquo;</a></nav>
          </article>
        </div>
      </section>
    </div>
  </div>
```

**Comment by jhabdas** [comment]
@liara looks good to me!

**Comment by liara** [comment]
Here you go @jhabdas :wink:

**Comment by jhabdas** [comment]
Closed @liara in #99. Changes available in `3.9.1`.
